src/tree/v2/ResourceBranchDataProviderManagerBase.ts

```diff
@@ -0,0 +1,78 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { BranchDataProvider, ResourceBase, ResourceModelBase } from '../../api/v2/v2AzureResourcesApi';
+
+export abstract class ResourceBranchDataProviderManagerBase<TBranchDataProvider extends BranchDataProvider<ResourceBase, ResourceModelBase>> extends vscode.Disposable {
+    private readonly branchDataProviderMap = new Map<string, { provider: TBranchDataProvider, listener: vscode.Disposable | undefined }>();
+    private readonly onDidChangeTreeDataEmitter = new vscode.EventEmitter<void | ResourceModelBase | ResourceModelBase[] | undefined | null>();
+
+    constructor(
+        private readonly defaultProvider: TBranchDataProvider,
+        private readonly extensionActivator: (type: string) => void
+    ) {
+        super(
+            () => {
+                this.onDidChangeTreeDataEmitter.dispose();
+
+                for (const providerContext of this.branchDataProviderMap.values()) {
+                    providerContext.listener?.dispose();
+                }
+            });
+    }
+
+    public readonly onDidChangeTreeData = this.onDidChangeTreeDataEmitter.event;
+
+    addProvider(type: string, provider: TBranchDataProvider): void {
+        type = this.normalizeType(type);
```

Comment:
`AzExtResourceType`!

Reply:
As mentioned above, I'd like to address this as a separate PR if necessary.  I'd like to make sure that I fully understand how the "normalized" `AzExtResourceType` works (e.g. for new resource types).